src/shared/types/props/StyleProps.type.ts

```diff
@@ -0,0 +1,70 @@
+import { CSSProperties } from 'react';
+
+export type StyleProps = FlexboxProps &
+  GridProps &
+  SpacingProps &
+  SizeProps &
+  TypographyProps &
+  BackgroundProps &
+  BoxModelProps &
+  PositionProps &
+  OtherProps;
+
+type PickCSSProperties<Properties extends keyof CSSProperties> = Partial<Pick<CSSProperties, Properties>>;
+
+export type FlexboxProps = PickCSSProperties<
+  | 'display'
+  | 'flex'
+  | 'flexDirection'
+  | 'justifyContent'
+  | 'alignItems'
+  | 'alignContent'
+  | 'flexWrap'
+  | 'flexBasis'
+  | 'flexGrow'
+  | 'flexShrink'
+  | 'order'
+  | 'alignSelf'
+  | 'gap'
+>;
+
+export type GridProps = PickCSSProperties<
+  | 'gridGap'
+  | 'gridRowGap'
+  | 'gridColumnGap'
+  | 'gridTemplateRows'
+  | 'gridTemplateColumns'
+  | 'gridTemplateAreas'
+  | 'gridArea'
```

Comment:
```suggestion
  | 'grid${'Gap'| 'RowGap' | '...'}'
```

이런식으로 타입을 선언할 수도 있답니다 ㅎㅎ

Reply:
> 이런식으로 타입을 선언할 수도 있답니다 ㅎㅎ

😲